workbench-app/Dockerfile

```
@@ -0,0 +1,39 @@
+# Stage 1: Build the React application
+FROM node:18-alpine AS build
+WORKDIR /app
+
+# Install pnpm
+RUN npm install -g pnpm
+
+# Copy package.json and pnpm-lock.yaml
+COPY package.json pnpm-lock.yaml ./
+
+# Install dependencies
+RUN pnpm install
+
+# Copy the rest of the application code
+COPY . .
+
+# Build the application
+RUN pnpm run build
```

on the subject of the `VITE_SEMANTIC_WORKBENCH_SERVICE_URL` - take a look at our current build process in `.github/workflows/workbench-app.yml`. you'll notice that it sets a number of environment variables at build time: `VITE_SEMANTIC_WORKBENCH_AUTHORITY`, `VITE_SEMANTIC_WORKBENCH_CLIENT_ID`, and `VITE_SEMANTIC_WORKBENCH_SERVICE_URL`.

considering this is now building the app code at docker-build time, i recommend using the same approach here with docker build args.

```suggestion
ARG VITE_SEMANTIC_WORKBENCH_AUTHORITY
ARG VITE_SEMANTIC_WORKBENCH_CLIENT_ID
ARG VITE_SEMANTIC_WORKBENCH_SERVICE_URL
RUN pnpm run build
```

with this change, the changes to the following files can be reverted:
- `workbench-app/index.html`
- `workbench-app/.env.example`
- `workbench-app/src/Constants.ts`
- `workbench-app/src/global.d.ts`

additionally, the `workbench-app/docker-entrypoint.sh` could be simplified to not both with `VITE_SEMANTIC_WORKBENCH_SERVICE_URL` at all.

---

Absolutely correct. I will implement this.